src/bidiMapper/domains/context/BrowsingContextImpl.ts

```diff
@@ -853,6 +909,94 @@ export class BrowsingContextImpl {
   }
 }
 
+type Rect = {
+  x: number;
+  y: number;
+  width: number;
+  height: number;
+};
+
+function deserializeDOMRect(rect: Script.EvaluateResult) {
+  if (
+    rect.type !== 'success' ||
+    rect.result.type !== 'object' ||
+    rect.result.value === undefined
+  ) {
+    return;
+  }
+  const x = rect.result.value.find(([key]) => {
+    return key === 'x';
+  })?.[1];
+  const y = rect.result.value.find(([key]) => {
+    return key === 'y';
+  })?.[1];
+  const height = rect.result.value.find(([key]) => {
+    return key === 'height';
+  })?.[1];
+  const width = rect.result.value.find(([key]) => {
+    return key === 'width';
+  })?.[1];
+  if (
+    x?.type !== 'number' ||
+    y?.type !== 'number' ||
+    height?.type !== 'number' ||
+    width?.type !== 'number'
+  ) {
+    return;
+  }
+  return {
+    x: x.value,
+    y: y.value,
+    width: width.value,
+    height: height.value,
+  } as Rect;
+}
+
+function normalizeRect(box: Readonly<Rect>): Rect {
```

Comment:
Comment here would be helpful as its hard to visualize what we are doing.
Maybe extracting them into two variable can help reduce the confusion as its very nested.

Reply:
Closed without a comment.